tools/regen_gh_pages.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+
+function _e() {
+    if [ ! -z "$VERBOSE" ]; then
+        echo cd "$(pwd)" '&&' "$@" 1>&2
+    fi
+
+    local command="$1"
+    shift
+
+    "$command" "$@"
+}
+
+function build_javadoc() {
+    local output_dir="$1"
+
+    if [ -e CMakeLists.txt ]; then
+        # Build Javadoc only
+        pushd build
+            _e cmake ..
+            _e make javadoc
+
+            # Preserve javadoc
+            _e cp -rv docs "$output_dir"
+        popd # build
+    else
+        local work_dir="/tmp/jss-docs-build"
+
+        _e rm -rf "$work_dir"
+        _e mkdir -p "$work_dir"
+
+        # Build the RPMs; easiest way to build the javadocs
+        _e bash ./build.sh --work-dir="$work_dir" rpm
+
+        # Extract the javadocs
+        pushd "$work_dir/RPMS"
+            _e rpm2cpio jss-javadoc*.rpm > javadoc.cpio
+            _e cpio -idv < javadoc.cpio
+            _e cp -rv ./usr/share/javadoc/jss* "$output_dir"
+        popd # "$work_dir"
+
+        rm -rf "$work_dir"
+    fi
+}
+
+function build_doc() {
+    local output_dir="$1"
+
+    if [ -e docs/ ]; then
+        _e cp -rv docs "$output_dir"
+    fi
+}
+
+function clean_checkout() {
+    local branch="$1"
+
+    _e git clean -xdf
+    _e git checkout "$branch"
+}
+
+function main() {(
+    set -e
+
+    local user="$1"
+    local remote="https://github.com/$user/jss"
+    local upstream="https://github.com/dogtagpki/jss"
+
+    local sandbox="/tmp/jss-docs-sandbox"
+    local artifacts="/tmp/jss-docs-artifacts"
+
+    if [ -z "$user" ]; then
+        echo "Usage: update-gh-pages.sh <username>"
+        echo "Update the user's gh-pages branch with the latest javadocs content."
+        return 1
+    fi
+
+    # Start with a fresh environment each time.
+    _e rm -rf "$sandbox" "$artifacts"
+    _e mkdir -p "$sandbox"
+    _e mkdir -p "$artifacts"
+
+    pushd "$sandbox"
+        _e git clone "$remote" jss
+        cd jss
+        _e git remote add upstream "$upstream"
+        _e git fetch --all
+
+        # Build docs for master branch
+        clean_checkout "upstream/master"
+        build_javadoc "$artifacts/javadocs-master"
+        build_doc "$artifacts/docs-master"
+
+        # Build docs for the v4.5.x branch
+        clean_checkout "upstream/v4.5.x"
+        build_javadoc "$artifacts/javadocs-v4.5.x"
+        build_doc "$artifacts/docs-v4.5.x"
+
+        # Build docs for the v4.4.x branch
```

Comment:
This patch is against `master` branch. Should we build for all branches?

Or my other concern is if I push a change to v4.4.x branch, will the CI run this script from master?

Reply:
I'm mostly placing this here as a placeholder for now. Currently:

- This script clones a git repo in a completely separate location
- This script builds all branches and updates accordingly

Because of that, master is the best place to put it, and it'd only be run from master. Logic is maintained in one place and that's it.

Ideally, this script would:

- Live in the otherwise-discussed automation pipeline and not in JSS.
- And yes, it'd support building just a single branch instead of all branches (actually, a fairly easy fix). 
- Use an existing git tree and only fetch the latest changes instead of everything.
- Support in-tree builds of JSS v4.4.x branch instead of pulling them from the javadocs RPM.